$MirOS: ports/www/firesomething/patches/patch-directory_c-sdk_configure_in,v 1.3 2008/04/02 00:35:17 tg Exp $
--- directory/c-sdk/configure.in.orig	Wed Feb 22 23:58:25 2006
+++ directory/c-sdk/configure.in	Tue May 20 00:20:55 2008
@@ -1569,7 +1569,23 @@ mips-sony-newsos*)
     _DEBUG_FLAGS="-gstabs"
 	;;
 
-*-openbsd*)
+*-midnightbsd*)
+    if test -z "$USE_NSPR_THREADS"; then
+        USE_PTHREADS=1
+    fi
+    AC_DEFINE(XP_UNIX)
+    AC_DEFINE(FREEBSD)
+    AC_DEFINE(HAVE_BSD_FLOCK)
+    CFLAGS="$CFLAGS $(DSO_CFLAGS) -ansi -Wall"
+    MOZ_OBJFORMAT=elf
+    DLL_SUFFIX=so
+    DSO_CFLAGS=-fPIC
+    DSO_LDOPTS=-Bshareable
+    MDCPUCFG_H=_freebsd.cfg
+    PR_MD_CSRCS=freebsd.c
+    ;;
+
+*-openbsd*|*-mirbsd*)
     AC_DEFINE(XP_UNIX)
     AC_DEFINE(OPENBSD)
     AC_DEFINE(HAVE_BSD_FLOCK)
@@ -2179,7 +2195,7 @@ if test -n "$USE_PTHREADS"; then
             _PTHREAD_LDFLAGS=
         fi
 	    ;;
-    *-freebsd*|*-openbsd*|*-bsdi*|*-netbsd*)
+    *-freebsd*|*-openbsd*|*-bsdi*|*-netbsd*|*-mirbsd*|*-midnightbsd*)
 	    AC_DEFINE(_THREAD_SAFE)
 	    dnl -pthread links in -lc_r, so don't specify it explicitly.
 	    if test "$ac_cv_have_dash_pthread" = "yes"; then
